annotated.go

```diff
@@ -519,16 +953,44 @@ func OnStart(onStart interface{}) Annotation {
 //	   func(lifecycle fx.Lifecycle, p Params) Server {
 //	     server := NewServer(p)
 //	     lifecycle.Append(fx.Hook{
-//		      OnStart: func(ctx context.Context) error {
+//		      OnStop: func(ctx context.Context) error {
 //			    return server.Shutdown(ctx)
 //		      },
 //	     })
+//		 return server
 //	   }
 //	 )
 //
+// It is also possible to use OnStop annotation with other parameter and result
+// annotations, provided that the parameter of the function passed to OnStop
+// matches annotated parameters and results
+//
+// For example, the following is possible:
+//
+//	fx.Provide(
+//		fx.Annotate(
+//			func (a A) B {...},
+//			fx.ParamTags(`name:"A"`),
+//			fx.ResultTags(`name:"B"`),
+//			fx.OnStop(func (p ParamStruct) {...}),
```

Comment:
nit: in the code sample, let's rename ParamStruct to OnStopParams

Reply:
Applied suggested change